packages/calypso-e2e/src/lib/pages/cart-checkout-page.ts

```diff
@@ -304,4 +308,27 @@ export class CartCheckoutPage {
 			this.page.click( selectors.purchaseButton ),
 		] );
 	}
+
+	/**
+	 * Complete the purchase accepting third party developer check and clicking on the 'Pay' button.
+	 */
+	async purchaseWithPlugin( { timeout }: { timeout?: number } = {} ): Promise< void > {
+		await this.page.getByLabel( selectors.thirdPartyDeveloperCheckboxLabel ).check();
+		await Promise.all( [
+			this.page.waitForResponse( /.*marketplace\/thank-you.*/, { timeout: timeout } ),
+			this.page.click( selectors.purchaseButton ),
+		] );
+	}
```

Comment:
[I checked the code](https://github.com/Automattic/wp-calypso/blob/ccac18fca861c5f812a1c0503528f295023bab28/client/my-sites/checkout/composite-checkout/components/wp-checkout.tsx#L482) and this checkbox appears when the cart has a Marketplace product, and in this case the `waitForResponse` URL is different. 
I will use your `Option 2` but instead of `checkThirdPartyAgree`, I will add `hasMarketplaceProduct`

Reply:
Btw, I will not use the purchase method until I can test it purchase a plugin (after my AFK)

Reply:
I created a follow-up for the purchase process here: https://github.com/Automattic/dotcom-forge/issues/2032